### PR TITLE
Airflow 3: python operators deprecations removal

### PR DESCRIPTION
**Pull request by dirrao**
Airflow 3: python operators deprecations removal

### CODE REVIEW

**Review by eladkal — CHANGES_REQUESTED** [review]
We need to wait with breaking changes to core operators/sensors

See
https://lists.apache.org/thread/2dmlqkcmyomm4q7rrovygs6bw655zx07

**Comment by dirrao** [comment]
> We need to wait with breaking changes to core operators/sensors
> 
> See https://lists.apache.org/thread/2dmlqkcmyomm4q7rrovygs6bw655zx07

Are we planning to introduce new provider in Airflow 3?

**Comment by jscheffl** [comment]
> > We need to wait with breaking changes to core operators/sensors
> > See https://lists.apache.org/thread/2dmlqkcmyomm4q7rrovygs6bw655zx07
> 
> Are we planning to introduce new provider in Airflow 3?

But... @eladkal we shoudl "clean-up" before carrying the legacy to new operators.. or? In my eyes we shoudl not carry-over deprecations.

**Comment by jscheffl** [comment]
Newsfragment is missing though :-D Please add...

**Comment by dirrao** [comment]
> Newsfragment is missing though :-D Please add...

Added the news fragment for this change.

**Comment by eladkal** [comment]
If we move the operator to providers there is no need for newsfragment. The breaking change will be in providers not in core.

File: newsfragments/41493.significant.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Removed deprecated method ``task`` from ``airflow.operators.python`` operator. Please use ``python_task`` from ``airflow.decorators.python`` instead.
```
Comment (eladkal) [inline newsfragments/41493.significant.rst:1]: The newsfragment will be about removing all the classses in the file.
This specific removal you are referring to will be in provider breaking change

Comment (potiuk) [reply newsfragments/41493.significant.rst:1]: I think we will have to keep it in newsfragments, so far those were "core" operators - and we have not moved them to "provider.standard" yet - and as I understand that, the standard provider will be part of the Airflow 3 migration (we will not remove old Python operators from Airflow 2 - we will add deprecation there)  so the newsfragment should be still in Airflow - I think.

**Comment by jscheffl** [comment]
I see two other deprecations in python.py - can you also remove these in the scope of the PR?

**Comment by dirrao** [comment]
> Newsfragment is missing though :-D Please add...

I have added the news fragment.

**Comment by dirrao** [comment]
> I see two other deprecations in python.py - can you also remove these in the scope of the PR?

Serialization code refactoring changes added recently (may be 2.10). Does it make sense to remove them immediately?

**Comment by jscheffl** [comment]
> > I see two other deprecations in python.py - can you also remove these in the scope of the PR?
> 
> Serialization code refactoring changes added recently (may be 2.10). Does it make sense to remove them immediately?

Deprecated is deprecated I'd say. It is by warning announced to be gone. Even if it is just announced in 2.10, this means 6+ months of notice before release. As well if we not remove in 3.0, then we would need to carry until 4.0

**Comment by dirrao** [comment]
> > > I see two other deprecations in python.py - can you also remove these in the scope of the PR?
> > 
> > 
> > Serialization code refactoring changes added recently (may be 2.10). Does it make sense to remove them immediately?
> 
> Deprecated is deprecated I'd say. It is by warning announced to be gone. Even if it is just announced in 2.10, this means 6+ months of notice before release. As well if we not remove in 3.0, then we would need to carry until 4.0

Yes. However, I want to reconfirm once before making the changes. 
@eladkal  / @potiuk WDYT?

**Comment by potiuk** [comment]
I think we need to wait for next dev call where we discuss "standard" provider's scope

**Comment by dirrao** [comment]
> I think we need to wait for next dev call where we discuss "standard" provider's scope

Any update on this?

**Comment by dirrao** [comment]
@potiuk / @eladkal 
are we good to merge this PR?

**Comment by potiuk** [comment]
> @potiuk / @eladkal are we good to merge this PR?

No - not really - there is another change in progress - https://github.com/apache/airflow/pull/42081 that moves Python operators to "standard" providers, and I think what you did here should be done as part of it, rather than separate PR from you - so I think the best course of action is to close that one and get you @dirrao to review and make comment in the #42081 so that we make sure deprecations are removed in the standard provider.

This should be generally done in the same way for all operators that are part of the "core" - we should move all of them to " "standard" and remove the deprecation "along the way". 

cc: @gopidesupavan.

**Pull request closed without merging**